gui/wxpython/gui_core/infobar.py

```diff
@@ -0,0 +1,89 @@
+
+import wx
+import wx.aui
+try:
+    import wx.lib.agw.infobar as IB
+except ImportError:
+    import wx.lib.infobar as IB
+
+
+class InfoBar(IB.InfoBar):
+    """
+    Custom Info bar
+    """
+    def __init__(self, parent):
+        IB.InfoBar.__init__(self, parent)
+
+        colBg = wx.SystemSettings.GetColour(wx.SYS_COLOUR_HIGHLIGHT)
+        self.SetBackgroundColour(colBg)
+        self.SetOwnForegroundColour(wx.SystemSettings.GetColour(wx.SYS_COLOUR_HIGHLIGHTTEXT))
+        self._text.SetBackgroundColour(colBg)
+        self._button.SetBackgroundColour(colBg)
+
+        # LAYOUT
+        sizer = wx.BoxSizer(wx.VERTICAL)
+        self.subSizerText = wx.BoxSizer(wx.HORIZONTAL)
+        self.subSizerButtons = wx.BoxSizer(wx.HORIZONTAL)
+        self.subSizerText.Add(self._icon, wx.SizerFlags().Centre().Border())
+        self.subSizerText.Add(self._text, 200, wx.ALIGN_CENTER_VERTICAL)
+        self.subSizerText.AddStretchSpacer()
+        self.subSizerText.Add(self._button, wx.SizerFlags().Centre().Border())
+        self.subSizerButtons.AddStretchSpacer()
+        sizer.Add(self.subSizerText, wx.SizerFlags().Expand())
+        sizer.Add(self.subSizerButtons, wx.SizerFlags().Expand())
+        self.SetSizer(sizer)
+
+    def AddButton(self, btnid, label, bitmap=wx.NullBitmap):
+        """
+        Adds a button to be shown in the info bar.
+        """
+        sizer = self.GetSizer()
+
+        assert sizer is not None, "Sizer must be created first"
+
+        # user-added buttons replace the standard close button so remove it if we
+        # hadn't done it yet
+        if sizer.Detach(self._button):
+            self._button.Hide()
+
+        button = wx.Button(self, btnid, label)
```

Comment:
Make the buttons less visible by using the same background color like the info bar (same as for the X button (`self._button`).

```suggestion
        button = wx.Button(self, btnid, label)
        button.SetBackgroundColour(self._background_color)
        button.SetForegroundColour(self._foreground_color)
```

This creates buttons with a thin okay outline for me. Hopefully, it looks okay elsewhere too.